### PR TITLE
Test if synchronization is disabled

### LINKED ISSUE #579: Syscheck automated tests: Synchronization disabled
## Description

Add test that checks that synchronization is disabled when set to disabled in the configuration.  
This message must not appear in the log:
```
2020/03/06 11:51:58 ossec-syscheckd[23205] fim_sync.c:56 at fim_run_integrity(): DEBUG: Initializing FIM Integrity Synchronization check. Sync interval is 300 seconds.
```
This is the required configuration for the synchronization:
```xml
<synchronization>
      <enabled>no</enabled>
      <interval>5m</interval>
      <max_interval>1h</max_interval>
      <max_eps>10</max_eps>
</synchronization>
```

### PR DESCRIPTION
**Pull request by jesusjimsa**
| Related issue  |
|---------------|
| #579                |

## Description

Add test to check if synchronization is disabled when `enabled` is set to `no` in the configuration:

```xml
<synchronization>
      <enabled>no</enabled>
      <interval>5m</interval>
      <max_interval>1h</max_interval>
      <max_eps>10</max_eps>
</synchronization>
```

Closes #579.

## Platforms supported by this test
- Linux
- Windows
- macOS

### CODE REVIEW

**Comment by jesusjimsa** [comment]
## Linux result
```
root@agentUbuntu19:/home/vagrant/qa/tests/integration/test_fim# python3 -m pytest test_sync/test_sync_disabled.py -vvs
========================================================= test session starts =========================================================
platform linux -- Python 3.7.3, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.7.3', 'Platform': 'Linux-5.0.0-36-generic-x86_64-with-Ubuntu-19.04-disco', 'Packages': {'pytest': '5.3.5', 'py': '1.8.1', 'pluggy': '0.13.1'}, 'Plugins': {'html': '2.0.1', 'metadata': '1.8.0'}}
rootdir: /home/vagrant/qa/tests/integration, inifile: pytest.ini
plugins: html-2.0.1, metadata-1.8.0
collected 1 item

test_sync/test_sync_disabled.py::test_sync_disabled[get_configuration0] 2020/03/06 12:54:09 ossec-syscheckd[5584] main.c:100 at main(): DEBUG: Starting ...
2020/03/06 12:54:09 ossec-syscheckd[5584] config.c:68 at Read_Syscheck_Config(): DEBUG: (6287): Reading configuration file: '/var/ossec/etc/ossec.conf'
2020/03/06 12:54:09 rootcheck[5584] rootcheck.c:197 at rootcheck_init(): DEBUG: Starting ...
PASSED2020-03-06T12:54:15.1583499255


========================================================== 1 passed in 6.46s ==========================================================
```

**Comment by jesusjimsa** [comment]
## Windows result

```
C:\Users\jmv74211\Desktop\wazuh-qa\tests\integration\test_fim> python -m pytest .\test_sync\test_sync_disabled.py -vvs
================================================= test session starts =================================================
platform win32 -- Python 3.7.3, pytest-5.1.2, py-1.8.0, pluggy-0.13.0 -- C:\Users\jmv74211\AppData\Local\Programs\Python
\Python37-32\python.exe
cachedir: .pytest_cache
metadata: {'Python': '3.7.3', 'Platform': 'Windows-7-6.1.7601-SP1', 'Packages': {'pytest': '5.1.2', 'py': '1.8.0', 'pluggy': '0.13.0'}, 'Plugins': {'html': '2.0.1', 'metadata': '1.8.0'}}
rootdir: C:\Users\jmv74211\Desktop\wazuh-qa\tests\integration, inifile: pytest.ini
plugins: html-2.0.1, metadata-1.8.0
collected 1 item

test_sync\test_sync_disabled.py::test_sync_disabled[get_configuration0] El servicio de Wazuh no se ha iniciado.

Puede obtener más ayuda con el comando NET HELPMSG 3521.

El servicio de Wazuh está iniciándose.
El servicio de Wazuh se ha iniciado correctamente.

PASSED
El servicio de Wazuh se detuvo correctamente.

El servicio de Wazuh está iniciándose.
El servicio de Wazuh se ha iniciado correctamente.



================================================= 1 passed in 17.40s ==================================================
```

File: tests/integration/test_fim/test_sync/test_sync_disabled.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import pytest
+
+from wazuh_testing import global_parameters
+from wazuh_testing.fim import LOG_FILE_PATH, callback_detect_synchronization, detect_initial_scan, generate_params
+from wazuh_testing.tools import PREFIX
+from wazuh_testing.tools.configuration import load_wazuh_configurations, check_apply_test
+from wazuh_testing.tools.file import truncate_file
+from wazuh_testing.tools.monitoring import FileMonitor
+from wazuh_testing.tools.time import TimeMachine, time_to_timedelta
```
Comment (vicferpoy) [inline tests/integration/test_fim/test_sync/test_sync_disabled.py]: Remove unused imports.

File: tests/integration/test_fim/test_sync/test_sync_disabled.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# Marks
+
+pytestmark = [pytest.mark.tier(level=2)]
```
Comment (vicferpoy) [inline tests/integration/test_fim/test_sync/test_sync_disabled.py]: Is this a tier 2 test?

Comment (jesusjimsa) [reply tests/integration/test_fim/test_sync/test_sync_disabled.py]: Not really, it should actually be a tier 1 test.

File: tests/integration/test_fim/test_sync/test_sync_disabled.py (hunk 1 above)
Comment (jesusjimsa) [reply tests/integration/test_fim/test_sync/test_sync_disabled.py]: Fixed 1838cc5

File: tests/integration/test_fim/test_sync/test_sync_disabled.py (hunk 2 above)
Comment (jesusjimsa) [reply tests/integration/test_fim/test_sync/test_sync_disabled.py]: Fixed 1838cc5

**Comment by jesusjimsa** [comment]
About [this comment](https://github.com/wazuh/wazuh-qa/pull/580#pullrequestreview-370428367), I think it should stay in `test_fim/test_sync`, because it's there where `synchronization` configuration is tested while in `test_fim/test_synchronize`, the functionality is tested.

**Comment by jesusjimsa** [comment]
Done d490356a754762e5a97b3db4fcba96ac067afb33

**Pull request merged**